cmd/delete.go

```diff
@@ -0,0 +1,54 @@
+package cmd
+
+import (
+	"errors"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/web-of-things-open-source/tm-catalog-cli/cmd/completion"
+	"github.com/web-of-things-open-source/tm-catalog-cli/internal/app/cli"
+	"github.com/web-of-things-open-source/tm-catalog-cli/internal/remotes"
+)
+
+var deleteCmd = &cobra.Command{
+	Use:   "delete <TMID>",
+	Short: "Delete a TM by id",
+	Long: `Delete a TM by id. Normally, the catalog is write-only and no TMs are ever deleted.
+The delete function is implemented for the rare cases when a TM has been pushed whilst containing major errors 
+or by mistake. Therefore, it is mandatory to provide the flag --force=true to delete a TM.`,
+	Args:              cobra.ExactArgs(1),
+	Run:               executeDelete,
+	ValidArgsFunction: nil, // just to make explicit in code that a completion function is not wanted here. Don't want to do anything to make deletion easier
+}
+
+func init() {
+	RootCmd.AddCommand(deleteCmd)
+	deleteCmd.Flags().StringP("remote", "r", "", "name of the remote to delete from")
+	_ = deleteCmd.RegisterFlagCompletionFunc("remote", completion.CompleteRemoteNames)
+	deleteCmd.Flags().StringP("directory", "d", "", "TM repository directory")
+	_ = deleteCmd.MarkFlagDirname("directory")
+	deleteCmd.Flags().StringP("force", "", "", "force the deletion") // intentionally a string flag, not boolean, so that the user has to make that much extra effort to type
```

Comment:
If we have no shorthand,, we should use the short version without `P` -> `deleteCmd.Flags().String(...`
In the other commands we should correct that than too...

Reply:
good point. changed to `String()`